esp/logging/logginglib/logconfigptree.hpp

```diff
@@ -0,0 +1,111 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _LOGCONFIGPTREE_HPP_
+#define _LOGCONFIGPTREE_HPP_
+
+#include "jptree.hpp"
+#include "jstring.hpp"
+#include "tokenserialization.hpp"
+
+namespace LogConfigPTree
+{
+    inline const char* queryProp(const IPTree& node, const char* xpath)
```

Comment:
Since this behavior is so different than other queryProp() type methods I would consider giving it a different name to avoid confusion.  Perhaps something like queryScalar() or queryScalarValue().  Unless you can think of something that intuitively means you'll look in a couple places or as Attr or El... if I could think of such a name I would give an example.